src/main/java/uk/gov/register/core/PostgresRegister.java

```diff
@@ -67,30 +67,33 @@ public void putItem(Item item) {
     public void appendEntry(final Entry entry) {
         List<Item> referencedItems = getReferencedItems(entry);
 
-        referencedItems.forEach(i -> {
+        referencedItems.forEach(item -> {
             if (entry.getEntryType() == EntryType.user) {
-                itemValidator.validateItem(i.getContent(), this.getFieldsByName(), this.getRegisterMetadata());
+                itemValidator.validateItem(item.getContent(), this.getFieldsByName(), this.getRegisterMetadata());
             } else if (entry.getKey().startsWith("field:")) {
-                Field field = extractObjectFromItem(i, Field.class);
+                Field field = extractObjectFromItem(item, Field.class);
                 environmentValidator.validateFieldAgainstEnvironment(field);
-                
+
             } else if (entry.getKey().startsWith("register:")) {
-                RegisterMetadata localRegisterMetadata = this.extractObjectFromItem(i, RegisterMetadata.class);
+                RegisterMetadata localRegisterMetadata = this.extractObjectFromItem(item, RegisterMetadata.class);
                 // will throw exception if field not present
                 localRegisterMetadata.getFields().forEach(this::getField);
-                
+
                 environmentValidator.validateRegisterAgainstEnvironment(localRegisterMetadata);
             }
         });
 
         entryLog.appendEntry(entry);
 
-        for (IndexFunction indexFunction : indexFunctions) {
-            indexDriver.indexEntry(this, entry, indexFunction);
-        }
-
         if (entry.getEntryType() == EntryType.user) {
+            for (IndexFunction indexFunction : indexFunctionsByEntryType.get(EntryType.user)) {
+                indexDriver.indexEntry(this, entry, indexFunction);
+            }
             recordIndex.updateRecordIndex(entry);
+        } else {
```

Comment:
We can remove this else and do the indexing as we know the entry type, then update the record index only if it's a user entry